### PR TITLE
view room

### PR DESCRIPTION
**Pull request by pntttt**

### CODE REVIEW

File: app/controllers/rooms_controller.rb [hunk 1]
```diff
@@ -1,7 +1,13 @@
 class RoomsController < ApplicationController
   before_action :load_room, except: [:index, :new, :create]
   before_action :authenticate_user!, except: [:show]
-  before_action :is_authorised?, except: [:index, :new, :create]
+  before_action :is_authorised?, except: [:index, :new, :create, :show]
+
+  def show
+    @room = Room.find_by id: params[:id]
```
Comment (thangtx88) [inline app/controllers/rooms_controller.rb]: lặp code

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -216,3 +216,17 @@ a {
...
+
+
+// ROOM SHOW
+.amenities li {
+  margin-bottom: 10px;
+  font-size: 16px;
+  list-style-type: none;
+}
+
+.amenities .text-line-through {
```
Comment (TranBaVinhSon) [inline app/assets/stylesheets/application.scss]: Move to line `222`

File: app/controllers/rooms_controller.rb [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+    redirect_to root_path, alert: t("noti_not_found") unless @room.active
```
Comment (TranBaVinhSon) [inline app/controllers/rooms_controller.rb]: What's happen when room is null ? use `@room&.active`

File: app/views/reservations/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          <label><%= t("check_out") %></label>
+          <%= f.text_field :end_date, readonly: true, placeholder: t("end_date"), class: "form-control datepicker", disabled: true %>
+        </div>
+      </div>
+      <h4 class="message-alert text-center"><span id="message"></span></h4>
+      <div id="preview" style="display: none">
+        <table class="reservation-table">
+          <tbody>
+            <tr>
+              <td><%= t("price") %></td>
```
Comment (TranBaVinhSon) [inline app/views/reservations/_form.html.erb]: remove ()

File: app/views/reservations/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+              <td class="text-right">$<%= @room.price %></td>
+            </tr>
+            <tr>
+              <td><%= t("night") %></td>
```
Comment (TranBaVinhSon) [inline app/views/reservations/_form.html.erb]: remove ()

File: app/views/reservations/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+              <td class="text-right">x <span id="reservation_nights"></span></td>
+            </tr>
+            <tr>
+              <td class="total"><%= t("total") %></td>
```
Comment (TranBaVinhSon) [inline app/views/reservations/_form.html.erb]: remove ()

File: app/views/reservations/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,41 @@
+<div class="panel panel-default">
+  <div class="panel-heading">
+    <span><i class="fa fa-bolt" style="color: #ffb400"></i> $<%= @room.price %></span>
+    <span class="pull-right"><%= t("per_night") %></span>
+  </div>
+  <div class="panel-body">
+    <%= form_for([@room, @room.reservations.new]) do |f| %>
+      <div class="row">
+        <div class="col-md-6">
+          <label><%= t("check_in") %></label>
```
Comment (TranBaVinhSon) [inline app/views/reservations/_form.html.erb]: remove ()

File: app/views/reservations/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,41 @@
+<div class="panel panel-default">
+  <div class="panel-heading">
+    <span><i class="fa fa-bolt" style="color: #ffb400"></i> $<%= @room.price %></span>
+    <span class="pull-right"><%= t("per_night") %></span>
```
Comment (TranBaVinhSon) [inline app/views/reservations/_form.html.erb]: remove ()

File: app/views/reservations/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,41 @@
+<div class="panel panel-default">
+  <div class="panel-heading">
+    <span><i class="fa fa-bolt" style="color: #ffb400"></i> $<%= @room.price %></span>
```
Comment (TranBaVinhSon) [inline app/views/reservations/_form.html.erb]: please don't put css code in html file

File: app/views/rooms/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+<br/>
+<div class="row">
+  <div class="col-md-8">
+    <div class="row">
+      <div class="col-md-8">
+        <h1><%= @room.name %></h1>
+        <%= @room.address %>
+      </div>
+      <div class="col-md-4 text-center">
+        <%= @room.user.name %>
```
Comment (TranBaVinhSon) [inline app/views/rooms/show.html.erb]: delegate -> `@room.user_name`

File: app/views/rooms/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+        <div class="col-md-3"><%= @room.home_type %></div>
+        <div class="col-md-3"><%= pluralize(@room.accommodate, t("guest")) %></div>
+        <div class="col-md-3"><%= pluralize(@room.bed_room, t("bedrooms")) %></div>
+        <div class="col-md-3"><%= pluralize(@room.bath_room, t("bathrooms")) %></div>
+      </div>
+    </div>
+    <hr/>
+    <div class="row">
+      <div class="col-md-12">
+        <h3><%= t("about_this_listing") %></h3>
```
Comment (TranBaVinhSon) [inline app/views/rooms/show.html.erb]: remove ()

File: app/views/rooms/show.html.erb [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+        <p><%= @room.summary %></p>
+      </div>
+    </div>
+    <hr/>
+    <div class="row">
+      <div class="col-md-3">
+        <h4><%= t("amenities") %></h4>
```
Comment (TranBaVinhSon) [inline app/views/rooms/show.html.erb]: remove ()

File: app/views/rooms/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+        </div>
+      <% end %>
+    </div>
+    <hr/>
+
+    <div class="row">
+      <h3><%= t("address") %></h3>
+      <div id="map" style="width: 100%; height: 400px"></div>
+      <script src="https://maps.googleapis.com/maps/api/js"></script>
+      <script>
```
Comment (TranBaVinhSon) [inline app/views/rooms/show.html.erb]: please don't write js code at html file. move to `.js` file

File: db/migrate/20180726080001_add_map_to_room.rb [hunk 14]
```diff
@@ -0,0 +1,6 @@
+class AddMapToRoom < ActiveRecord::Migration[5.1]
+  def change
+    add_column :rooms, :latitude, :float
```
Comment (TranBaVinhSon) [inline db/migrate/20180726080001_add_map_to_room.rb]: add `null: false`

File: app/controllers/reservations_controller.rb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+class ReservationsController < ApplicationController
```
Comment (TranBaVinhSon) [inline app/controllers/reservations_controller.rb]: Why do you need this file ?

File: config/routes.rb [hunk 16]
```diff
@@ -16,5 +16,6 @@
       get "photo_upload"
     end
     resources :photos, only: [:create, :destroy]
+    resources :reservations
```
Comment (TranBaVinhSon) [inline config/routes.rb]: Did you use all 7 functions ?

**Pull request merged**